protocols/directfund/directfund.go

```diff
@@ -203,6 +215,49 @@ func (o Objective) Channels() []*channel.Channel {
 	return ret
 }
 
+// MarshalJSON returns a JSON representation of the DirectFundObjective
+//
+// NOTE: Marshal -> Unmarshal is a lossy process. All channel data
```

Comment:
Having `Marshal -> Unmarshal` being lossy doesn't seem the best.

I wonder if it would make sense to move MarshalJSON/UnMarshalJSON to the `jsonObjective` struct. Instead of an `Objective` marshalling data to json itself we'd just transform it into `jsonObjective`. That way we separate the problem of marshalling data from the problem of constructing an `Objective`  with fully defined channels.

Reply:
It definitely isn't the best!

The compromise is motivated by us wanting to have channels attached to objectives in memory, as part of the `update` `crank` loop, but we want to persist channels / objectives separately in the store.

Ideally, the marshalling / unmarshalling functionality would live, encapsulated, in the `store`. Issue there is that there are private fields in Objective & Channel which we need to persist, so they aren't visible inside `store`.